### DISCUSSION TITLE
Missing future energy prices...

### DISCUSSION THREAD

**Discussion by atle-dahlman** [UtilitechAS/amsreader-firmware#579] [General]
Hmmm, anyone else missing the graph in future energy prices?
Looks like it started at midnight May 28'th.

![image](https://github.com/UtilitechAS/amsreader-firmware/assets/101756371/3b8b526f-f1dd-4f9a-bde8-9c80542b00d7)

**Comment by gskjold** [comment]
Yes, entso-e is currently missing prices in their api

**Reply by dimnek** [reply]
I fetch the prices from entso-e to HA directly (not using the price feature of the AMS reader), and also noticed that they were missing this morning.

So I started to look for a backup solution and came across the Danish [Energi Data Service](https://www.energidataservice.dk/guides/api-guides) from Energinet. Energinet is an independent public enterprise owned by the Danish Ministry of Climate and Energy. The data is licensed as [Conditions for use of Danish public sector data](https://www.energidataservice.dk/Conditions_for_use_of_Danish_public_sector_data-License_for_use_of_data_in_ED.pdf). 

So far it looks promising, the danes have the prices (also for NO) even when entso-e does not.

So now I try entso-e first and fall back to Energi Data Service when I don't get what I want. That might be an idea for this project as well.

**Reply by gskjold** [reply]
Thank you for making me aware of this, I will implement this as a fallback for now. Also appreciate that you see the importance in the Terms of use of such a service
